### ISSUE TITLE
Performance diversion between RNN in Keras and Axon

### ISSUE THREAD

**Issue by seanmor5** [elixir-nx/axon#530]
There is a model performance divergence between RNNs in Keras and Axon.

The same RNN performs significantly better in Keras. Axon RNNs with `unroll: :dynamic` are not trainable at the moment because of issues with gradients. `unroll: :static` is better, but results in slow compiles and still has issues with training stability.
